File: kids_with_candies_high.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int max_candy = *max_element(candies.begin(),candies.end());
        vector<bool> result(candies.size());
        transform(candies.begin(),candies.end(),result.begin(),[max_candy,extraCandies](int i){
            return i + extraCandies >= max_candy;
        });
        return result;
    }
};

File: kids_with_candies_high.py
class Solution(object):
    def kidsWithCandies(self, candies, extraCandies):
        """
        :type candies: List[int]
        :type extraCandies: int
        :rtype: List[bool]
        """

        max_candy = max(candies)
        return list(map(lambda x : x + extraCandies >= max_candy,candies))
